HashTagTokenizer.java

```diff
@@ -6,35 +6,66 @@ public static void main(String[] args) {
 
 		String hashTag = args[0];
 		String []dictionary = readDictionary("dictionary.txt");
+		System.out.println (existInDictionary(hashTag, dictionary));
+		hashTag = hashTag.toLowerCase();
 		breakHashTag(hashTag, dictionary);
 	}
 
 	public static String[] readDictionary(String fileName) {
 		String[] dictionary = new String[3000];
 
 		In in = new In(fileName);
-
+		for (int i=0;i<dictionary.length-1; i++)
+			{
+				if (in != null && !in.isEmpty())
+				{
+					String word = in.readLine();
+					dictionary[i]= word;
+				}
```

Comment:
At the cases you mentioned in `if` it might better to break the loop when it happen instead of continue the iterations and found that condition is always `false`.
You could done it like this:
```suggestion
				if (in == null || in.isEmpty())
				{
					break;
				}
				String word = in.readLine();
				dictionary[i]= word;
```